flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2711,6 +2717,28 @@ class ScalarExprLowering {
         continue;
       }
       const bool actualArgIsVariable = Fortran::evaluate::IsVariable(*expr);
+      if (arg.passBy == PassBy::BaseAddressValueAttribute) {
+        mlir::Value temp;
+        if (isArray(*expr)) {
+          auto val = genBoxArg(*expr);
+          if (!actualArgIsVariable)
+            temp = getBase(val);
+          else {
+            ExtValue copy = genArrayTempFromMold(val, ".copy");
+            genArrayCopy(copy, val);
+            temp = fir::getBase(copy);
+          }
+        } else {
+          mlir::Value val = fir::getBase(genval(*expr));
+          temp = builder.createTemporary(
+              loc, val.getType(),
+              llvm::ArrayRef<mlir::NamedAttribute>{
+                  Fortran::lower::getAdaptToByRefAttr(builder)});
+          builder.create<fir::StoreOp>(loc, val, fir::getBase(temp));
+        }
+        caller.placeInput(arg, fir::getBase(temp));
```

Comment:
No need to call `fir::getBase` on something that is already an mlir::Value (calling it creates an fir::ExtendedValue from the Value that is passed to getBase, and then extract that Value back.)